### ISSUE TITLE
Android Studio can't compile ofxPd - Maybe be OF config issue

### ISSUE THREAD

**Issue by cerupcat** [openframeworks/openFrameworks#6251]
I'm trying to add ofxPd to project. I'm cross-posting here since the ofxPd developer and I think this could be a core issue. https://github.com/danomatika/ofxPd/issues/70

Reproduction:
```
Download latest Android OF version from website
Clone ofxPD to addon folder
Use project generator to import pdExample
Open generated Android project pdExample
Build
```

Error
```C compiler failed while compiling s_libpdmidi.c.```

It appears if I add to the build.gradle file `CFlags.addAll(addonCppFlags(abi, ofRoot()))` after `cppFlags.addAll(addonCppFlags(abi, ofRoot()))` this particular error goes away. It does appear that not all CFlags/CPPFlags are getting set though.

It's hard to say if the build issue is related to ofxPd or changes in OF. Any insights would help.

**Comment by cuinjune** [comment]
It seems like `ADDON_CFLAGS` doesn't work properly when the flags are assigned as a list.

It seems to work when I assign the flags like this instead:

```
    ADDON_CFLAGS = -DHAVE_UNISTD_H 
    ADDON_CFLAGS += -DUSEAPI_DUMMY
    ADDON_CFLAGS += -DPD
    ADDON_CFLAGS += -DLIBPD_EXTRA
    ADDON_CFLAGS += -DLIBPD_USE_STD_MUTEX
```

**Comment by danomatika** [comment]
Ping @arturoc. If only the first define makes it through from the list but individual append lines work, this seems like parsing issue.

**Comment by arturoc** [comment]
this is correct, in makefile syntax = overwrites the previous value while += adds to the previous value so that's the correct way to add more than one value. 

going to close this as i think it's correct as it is but let me know if i'm missing something

**Comment by danomatika** [comment]
I don’t recall makefile syntax doing the following:

CFLAGS = -DONE -DTWO

resulting in only -DONE affecting the build while dropping anything afterwards, which is the issue at hand.

enohp ym morf tnes
-----------
Dan Wilcox
danomatika.com
robotcowboy.com

**Comment by arturoc** [comment]
oh i see sorry i thought it didn't work when doing 

```
    ADDON_CFLAGS = -DHAVE_UNISTD_H 
    ADDON_CFLAGS = -DUSEAPI_DUMMY
    ADDON_CFLAGS = -DPD
    ADDON_CFLAGS = -DLIBPD_EXTRA
    ADDON_CFLAGS = -DLIBPD_USE_STD_MUTEX
```

this is a bug in the android build system then
